web-bluetooth/read-descriptors-async-await.js

```diff
@@ -0,0 +1,76 @@
+async function onButtonClick() {
+  let serviceUuid = document.querySelector('#service').value;
+  if (serviceUuid.startsWith('0x')) {
+    serviceUuid = parseInt(serviceUuid);
+  }
+
+  let characteristicUuid = document.querySelector('#characteristic').value;
+  if (characteristicUuid.startsWith('0x')) {
+    characteristicUuid = parseInt(characteristicUuid);
+  }
+
+  try {
+    log('Requesting any Bluetooth Device...');
+    const device = await navigator.bluetooth.requestDevice({
+     // filters: [...] <- Prefer filters to save energy & show relevant devices.
+        acceptAllDevices: true, optionalServices: [serviceUuid]});
+
+    log('Connecting to GATT Server...');
+    const server = await device.gatt.connect();
+
+    log('Getting Service...');
+    const service = await server.getPrimaryService(serviceUuid);
+
+    log('Getting Characteristic...');
+    const characteristic = await service.getCharacteristic(characteristicUuid);
+
+    log('Getting Descriptors...');
+    const descriptors = await characteristic.getDescriptors();
+
+    for (const descriptor of descriptors) {
+      switch (descriptor.uuid) {
+
+        case BluetoothUUID.getDescriptor('gatt.client_characteristic_configuration'):
+          await descriptor.readValue().then(value => {
+            log('> Client Characteristic Configuration:');
+            log('  > Notifications: ' + (value.getUint8(0) ? 'ON' : 'OFF'));
+            log('  > Indications: ' + (value.getUint8(1) ? 'ON' : 'OFF'));
```

Comment:
This is fetching the 2nd byte, but you intended to select the 2nd bit. Use something like:
 value.getUint8(0) & 0b01 // notifications
 value.getUint8(0) & 0b10 // indications